### ISSUE TITLE
Make decision tree classifier training deterministic

### ISSUE THREAD

**Issue by dalelane** [IBM/taxinomitis#229]
from @kevinrobinson in https://github.com/IBM/taxinomitis/issues/226 

> Relatedly, it looks like subsequent runs are non-deterministic [link](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html):
> 
>> The features are always randomly permuted at each split. Therefore, the best found split may vary, even with the same training data and max_features=n_features, if the improvement of the criterion is identical for several splits enumerated during the search of the best split. To obtain a deterministic behaviour during fitting, random_state has to be fixed.
> 
> This might be good to either add a line about in the explanation, or to avoid this to simplify the experience and fix a random seed so that folks aren't confused if they get different trees from different training runs on the same data set (eg, within a class).
